connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2724,8 +2724,20 @@ class Dataset[T] private[sql] (
     throw new UnsupportedOperationException("localCheckpoint is not implemented.")
   }
 
+  /**
+   * Returns `true` when the logical query plans inside both [[Dataset]]s are equal and
+   * therefore return same results.
+   *
+   * @note The equality comparison here is simplified by tolerating the cosmetic differences
+   *       such as attribute names.
+   * @note This API can compare both [[Dataset]]s very fast but can still return `false` on
```

Comment:
Update this comment? The comparison won't be fast. Maybe add a note here to explain that this executes a RPC.

Reply:
I see. gRPC might still fast, but just not as fast as before. I removed `very fast` and explain that it will execute a RPC call in comment